auth0/resource_auth0_log_stream.go

```diff
@@ -0,0 +1,330 @@
+		m = flattenLogStreamEventGridSink(o)
+	case *management.HTTPSink:
+		m = flattenLogStreamHTTPSink(o)
+	case *management.DatadogSink:
+		m = flattenLogStreamDatadogSink(o)
+	case *management.SplunkSink:
+		m = flattenLogStreamSplunkSink(o)
+	}
+	return []interface{}{m}
+}
+
+func flattenLogStreamEventBridgeSink(o *management.EventBridgeSink) interface{} {
+	return map[string]interface{}{
+		"aws_account_id":           o.GetAWSAccountID(),
+		"aws_region":               o.GetAWSRegion(),
+		"aws_partner_event_source": o.GetAWSPartnerEventSource(),
+	}
+}
+
+func flattenLogStreamEventGridSink(o *management.EventGridSink) interface{} {
+	return map[string]interface{}{
+		"azure_subscription_id": o.GetAzureSubscriptionID(),
+		"azure_resource_group":  o.GetAzureResourceGroup(),
+		"azure_region":          o.GetAzureRegion(),
+		"azure_partner_topic":   o.GetAzurePartnerTopic(),
+	}
+}
+
+func flattenLogStreamHTTPSink(o *management.HTTPSink) interface{} {
+	return map[string]interface{}{
+		"http_endpoint":       o.GetHTTPEndpoint(),
+		"http_contentFormat":  o.GetHTTPContentFormat(),
+		"http_contentType":    o.GetHTTPContentType(),
+		"http_authorization":  o.GetHTTPAuthorization(),
+		"http_custom_headers": o.HTTPCustomHeaders,
+	}
+}
+
+func flattenLogStreamDatadogSink(o *management.DatadogSink) interface{} {
+	return map[string]interface{}{
+		"datadog_region":  o.GetDatadogRegion(),
+		"datadog_api_key": o.GetDatadogAPIKey(),
+	}
+}
+
+func flattenLogStreamSplunkSink(o *management.SplunkSink) interface{} {
+	return map[string]interface{}{
+		"splunk_domain": o.GetSplunkDomain(),
+		"splunk_token":  o.GetSplunkToken(),
+		"splunk_port":   o.GetSplunkPort(),
+		"splunk_secure": o.GetSplunkSecure(),
+	}
+}
+func expandLogStream(d ResourceData) *management.LogStream {
+
+	c := &management.LogStream{
+		Name:   String(d, "name", IsNewResource()),
+		Type:   String(d, "type", IsNewResource()),
+		Status: String(d, "status"),
+	}
+
+	s := d.Get("type").(string)
+
+	List(d, "sink").Elem(func(d ResourceData) {
+		switch s {
+		case management.LogStreamSinkEventBridge:
+			c.Sink = expandLogStreamEventBridgeSink(d)
+		case management.LogStreamSinkEventGrid:
+			c.Sink = expandLogStreamEventGridSink(d)
+		case management.LogStreamSinkHTTP:
+			c.Sink = expandLogStreamHTTPSink(d)
+		case management.LogStreamSinkDatadog:
+			c.Sink = expandLogStreamDatadogSink(d)
+		case management.LogStreamSinkSplunk:
+			c.Sink = expandLogStreamSplunkSink(d)
+		default:
+			log.Printf("[WARN]: Raise an issue with the auth0 provider in order to support it:")
+			log.Printf("[WARN]: 	https://github.com/alexkappa/terraform-provider-auth0/issues/new")
+		}
+	})
+
+	return c
+}
+
+func expandLogStreamEventBridgeSink(d ResourceData) *management.EventBridgeSink {
+	o := &management.EventBridgeSink{
+		AWSAccountID:          String(d, "aws_account_id"),
+		AWSRegion:             String(d, "aws_region"),
+		AWSPartnerEventSource: String(d, "aws_partner-event_source"),
```

Comment:
Is the hyphen intentional?  On line 217 you've used `aws_partner_event_source` but here you've got `aws_partner-event_source`.

Reply:
Good catch, I've fixed it.

Reply:
👍